scoap3/modules/records/util.py

```diff
@@ -132,17 +128,12 @@ def get_abbreviated_journal(record):
 
 
 def get_arxiv_primary_category(record):
-    if "report_numbers" in record:
-        for i, report_number in enumerate(record["report_numbers"]):
-            if report_number['source'].lower() == 'arxiv':
-                if 'primary_category' not in report_number:
-                    # Since we added this field later, there can be records without it.
-                    # Ff we haven't extracted the primary_cetegory yet, do it now.
-                    arxiv_id = report_number['value'].lower().replace('arxiv:', '').split('v')[0]
-                    record["report_numbers"][i]['primary_category'] = get_arxiv_primary_category_by_id(arxiv_id)
-                    record.commit()
-
-                return record["report_numbers"][i]['primary_category']
+    """Return primary arXiv category for record. None if not available."""
+    for arxiv_eprints in record.get('arxiv_eprints', ()):
+        if 'categories' in arxiv_eprints and arxiv_eprints['categories']:
+            return arxiv_eprints['categories'][0]
+
+    return None
```

Comment:
using `get_value` here becomes simply
`get_value(record, 'arxiv_eprints.categories[0]')` and if there's no category, just returns `None`